src/assetContainer.ts

```diff
@@ -469,6 +469,7 @@ export class AssetContainer extends AbstractScene {
 
             if (move) {
                 targetAssets.push(asset);
+                (asset as any)._parentContainer = this;

```

Comment:
I wonder if we could not have something like:
```javascript
class AssetStorable {
    /** @hidden */
    public_parentContainer: AssetContainer;

    public removeFromContainer(arrayName: string): void {
        // we could also pass the arrayName in the constructor of the class and keep it as a property
        // remove from this._parentContainer[arrayName]
    }
}
```
That way we could declare:
```typescript
private _moveAssets<T extends AssetStorable>(sourceAssets: T[], targetAssets: T[], keepAssets: T[]): void {
```
instead of
```typescript
private _moveAssets<T>(sourceAssets: T[], targetAssets: T[], keepAssets: T[]): void {
```
and no need to cast to any in `(asset as any)._parentContainer = this;`.

That would avoid duplicating numerous times the same code but would mean all classes that can be stored in a asset container must extend `AssetStorable`.

Reply:
I like the idea but I will keep it for another PR